lightning/src/routing/scoring.rs

```diff
@@ -717,15 +735,34 @@ impl<G: Deref<Target = NetworkGraph<L>>, L: Deref, T: Time> ProbabilisticScorerU
 	/// Note that this writes roughly one line per channel for which we have a liquidity estimate,
 	/// which may be a substantial amount of log output.
 	pub fn debug_log_liquidity_stats(&self) {
+		let now = T::now();
+
 		let graph = self.network_graph.read_only();
 		for (scid, liq) in self.channel_liquidities.iter() {
 			if let Some(chan_debug) = graph.channels().get(scid) {
 				let log_direction = |source, target| {
 					if let Some((directed_info, _)) = chan_debug.as_directed_to(target) {
 						let amt = directed_info.effective_capacity().as_msat();
 						let dir_liq = liq.as_directed(source, target, amt, &self.params);
-						log_debug!(self.logger, "Liquidity from {:?} to {:?} via {} is in the range ({}, {})",
-							source, target, scid, dir_liq.min_liquidity_msat(), dir_liq.max_liquidity_msat());
+
+						let buckets = HistoricalMinMaxBuckets {
+							min_liquidity_offset_history: &dir_liq.min_liquidity_offset_history,
+							max_liquidity_offset_history: &dir_liq.max_liquidity_offset_history,
+						};
+						let (min_buckets, max_buckets, _) = buckets.get_decayed_buckets(now,
+							*dir_liq.last_updated, self.params.historical_no_updates_half_life);
+
+						log_debug!(self.logger, core::concat!(
+							"Liquidity from {:?} to {:?} via {} is in the range ({}, {}).\n",
```

Comment:
Loosely related: Could we `impl Display for NodeId`, or is it omitted on purpose? Might be generally useful and would make this a bit more concise?